src/Symfony/Component/HttpKernel/CacheWarmer/CacheWarmerAggregate.php

```diff
@@ -46,15 +50,62 @@ public function enableOnlyOptionalWarmers()
      */
     public function warmUp($cacheDir)
     {
-        foreach ($this->warmers as $warmer) {
-            if (!$this->optionalsEnabled && $warmer->isOptional()) {
-                continue;
-            }
-            if ($this->onlyOptionalsEnabled && !$warmer->isOptional()) {
-                continue;
+        if ($this->debug) {
+            $collectedLogs = array();
+            $previousHandler = defined('PHPUNIT_COMPOSER_INSTALL');
+            $previousHandler = $previousHandler ?: set_error_handler(function ($type, $message, $file, $line) use (&$collectedLogs, &$previousHandler) {
+                if (E_USER_DEPRECATED !== $type && E_DEPRECATED !== $type) {
+                    return $previousHandler ? $previousHandler($type, $message, $file, $line) : false;
+                }
+
+                if (isset($collectedLogs[$message])) {
+                    ++$collectedLogs[$message]['count'];
+
+                    return;
+                }
+
+                $backtrace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 3);
+                // Clean the trace by removing first frames added by the error handler itself.
+                for ($i = 0; isset($backtrace[$i]); ++$i) {
+                    if (isset($backtrace[$i]['file'], $backtrace[$i]['line']) && $backtrace[$i]['line'] === $line && $backtrace[$i]['file'] === $file) {
+                        $backtrace = array_slice($backtrace, 1 + $i);
+                        break;
+                    }
+                }
+
+                $collectedLogs[$message] = array(
+                    'type' => $type,
+                    'message' => $message,
+                    'file' => $file,
+                    'line' => $line,
+                    'trace' => $backtrace,
+                    'count' => 1,
+                );
+            });
+        }
+
+        try {
+            foreach ($this->warmers as $warmer) {
+                if (!$this->optionalsEnabled && $warmer->isOptional()) {
+                    continue;
+                }
+                if ($this->onlyOptionalsEnabled && !$warmer->isOptional()) {
+                    continue;
+                }
+
+                $warmer->warmUp($cacheDir);
             }
+        } finally {
+            if ($this->debug && true !== $previousHandler) {
```

Comment:
I've also prepare a test scenario with a cacheWarmer who throw deprecations but here I'm directly in conflict with https://github.com/symfony/symfony/pull/25413 (fixing #24767). It can't pass in this condition during the test and it's right.

Here was my test code (CacheWarmerAggregateTest):
```php
    public function testWarmupWithDeprecationLogs()
    {
        $cacheDir = self::$cacheDir;
        $warmer = $this->getCacheWarmerMock();
        $warmer
            ->expects($this->once())
            ->method('warmUp')
            ->with($cacheDir)
            ->will($this->returnValue($this->throwDeprecation()));
        ;

        $deprecationLogsFilepath = self::$cacheDir.'/Deprecations.log';

        $aggregate = new CacheWarmerAggregate(array($warmer), true, $deprecationLogsFilepath);
        $aggregate->warmUp($cacheDir);

        $this->assertTrue(file_exists($deprecationLogsFilepath));
        $this->assertCount(1, unserialize(file_get_contents($deprecationLogsFilepath)));
    }

    protected function throwDeprecation()
    {
        @trigger_error('Deprecation Message', E_USER_DEPRECATED);
    }
```

Reply:
Unsetting the en car temporarily should be doable?

Reply:
I've try many solution, and even temporarily removing the defined constant (runkit_constant_remove), deprecations have a different behavior during phpunit test. Still looking to find a solution and test it.

Reply:
ok, let's forget about it :):